networkx/algorithms/distance_measures.py

```diff
@@ -163,7 +104,20 @@ def _extrema_bounding(G, compute="diameter"):
         high = not high
 
         # get distances from/to current node and derive eccentricity
-        dist = dict(nx.single_source_shortest_path_length(G, current))
+        dist = nx.single_source_bellman_ford_path_length(
+            G=G, source=current, weight=weight
+        )
+        if isinstance(weight, str):
+            edge_weights = nx.get_edge_attributes(G, "weight").values()
+        else:
+            edge_weights = [weight(v, u, e) for u, e in G_succ[n].items()]
+
+        decimal_places = [str(w)[::-1].find(".") for w in edge_weights if w]
+        if decimal_places:
+            decimal_places = max(decimal_places)
+            for k in dist.keys():
+                dist[k] = round(dist[k], decimal_places)
+
```

Comment:
> Am I correct in inferring that this code is only needed so we can round off the distances to a certain number of decimal places?

You are correct.

> If so, we should leave this out. 

OK, this makes sense. Removing the precision-handling will cause issues with the existing tests, but I can address this by giving the tests integer weights.

> I like the way you figured out the number of decimals for the incoming weights though. :)

Thanks! :)

> And anyone using floating point weights will have to deal with round-off error on their own. 

Do you think it would make sense to add a note to this effect in the docstring?

Reply:
I guess it never hurts to remind people about round-off error. Something short -- and maybe where the `weight` input is described. like: "Weights stored as floating point values can lead to small round-off errors in distances. Use integer weights to avoid this."

For the tests, I think you should leave the tests in place testing the round-off error. For each failing assertion, change it to expect the failure (`assert ... != ...` or `assert not ...`). Then right below it add a line that tests that they are close in value. The [pytest utility approx](https://docs.pytest.org/en/latest/reference/reference.html?highlight=approx#pytest.approx) does a sophisticated checking process so you can do `assert expected == pytest.approx(actual)`.  And it checks all the values when you give it lists or dicts.  Anyway, test that round-off is a problem so we can see when that is no longer the case, and also test that the result is "close".

So: `assert nx.eccentricity(self.G, 1) == 15`
would become:
```python
assert nx.eccentricity(self.G, 1) != 15
assert pytest.approx(nx.eccentricity(self.G, 1)) == 15
```

Reply:
> The [pytest utility approx](https://docs.pytest.org/en/latest/reference/reference.html?highlight=approx#pytest.approx) does a sophisticated checking process so you can do assert expected == pytest.approx(actual)

Ooh, I was not aware of this - very cool!

I'll revert to the original float-weighted tests and make this adjustment.